docs/modules/ROOT/pages/upgrades.adoc

```diff
@@ -68,46 +65,29 @@ mod UpgradeableContract {
-        // This function can only be called by the owner
-        let ownable_state = Ownable::unsafe_new_contract_state();
-        Ownable::InternalImpl::assert_only_owner(@ownable_state);
-
-        // Replace the class hash upgrading the contract
-        let mut upgradeable_state = Upgradeable::unsafe_new_contract_state();
-        Upgradeable::InternalImpl::upgrade(ref upgradeable_state, new_class_hash);
+    impl UpgradeableImpl of IUpgradeable<ContractState> {
+        fn upgrade(ref self: ContractState, new_class_hash: ClassHash) {
+            // This function can only be called by the owner
+            let ownable_state = Ownable::unsafe_new_contract_state();
+            Ownable::InternalImpl::assert_only_owner(@ownable_state);
+
+            // Replace the class hash upgrading the contract
+            let mut upgradeable_state = Upgradeable::unsafe_new_contract_state();
+            Upgradeable::InternalImpl::_upgrade(ref upgradeable_state, new_class_hash);
+        }
     }
 
     (...)
 }
 ----
 
-=== Interface
-
-Implementing this interface to support upgradeability, supports interoperability by for example allowing DAOs
-and plugins to connect to contracts and upgrade them.
-
-[,javascript]
-----
-trait IUpgradeable {
-    fn upgrade(new_class_hash: ClassHash);
-}
-----
-
 == Proxies and Starknet
 
-A proxy contract is a contract that delegates function calls to another contract.
-This type of pattern decouples state and logic.
-Proxy contracts store the state and redirect function calls to an implementation contract that handles the logic,
-usually catching these calls with a fallback mechanism.
-This allows for different patterns such as upgrades, where implementation contracts can change but the proxy contract (and thus the state) does not;
-as well as deploying multiple proxy instances pointing to the same implementation.
-This can be useful to deploy many contracts with identical logic but unique initialization data.
+Proxies allow for different patterns such as upgrades as well as deploying multiple clones of the same contract. But Starknet achieves these two things in different ways.
 
-In the case of contract upgrades, it is achieved by simply changing the proxy's reference to the class hash of the declared implementation.
-This allows developers to add features, update logic, and fix bugs without touching the state or the contract address to interact with the application.
+In the case of contract upgrades, it is achieved by simply changing the contract's class hash. As of clones, contracts already are like clones of the class they implement.
 
-In Starknet, implementing a proxy pattern has an important limitation: there is no fallback mechanism to be used
-for redirecting every potential function call to the implementation contract. This means a generic proxy contract
+Implementing a proxy pattern in Starknet has an important limitation: there is no fallback mechanism to be used
```

Comment:
```suggestion
Implementing a proxy pattern in Starknet is not supported since there is no fallback mechanism to be used
```

Reply:
It is supported, the lack of a fallback mechanism doesn't imply a lack of support for proxies, as with the examples I've shared.